src/foam/dao/ArrayDAO.js

```diff
@@ -29,6 +29,14 @@ foam.CLASS({
 
   properties: [
     {
+      class: 'Class',
+      name: 'of',
+      factory: function() {
+        if (this.array.length === 0) return foam.lookup('foam.core.FObject');
```

Comment:
I thought it was standard practice. I suppose it should be `imports: ['lookup']` and `this.lookup()`.